lib/types/src/schema/eslint.ts

```diff
@@ -1,10 +1,10 @@
-import { SchemaOutput } from '../schema'
+import { z } from 'zod'
 
-export const ESLintSchema = {
-  files: 'array.string',
-  config: 'record.unknown?', // @deprecated: use options instead
-  options: 'record.unknown?'
-} as const
-export type ESLintOptions = SchemaOutput<typeof ESLintSchema>
+export const ESLintSchema = z.object({
+  files: z.string().array().default(['**/*.js']),
+  config: z.record(z.unknown()).optional(), // @deprecated: use options instead
+  options: z.record(z.unknown()).optional()
+})
+export type ESLintOptions = z.infer<typeof ESLintSchema>
```

Comment:
**thought:** I like the structure we get in the schema here - specifying types as well as defaults and optional values. I don't love the abstraction of the options using `z.infer` because it is additional library code to mentally parse but I understand it and think the descriptive benefits we get in the schema are worth it.

Reply:
Imo it isn't any worse than the `SchemaOutput` type we were using previously to extract the `*Options` types. And helpfully in both cases, VSCode will also show you what the resolved type is if you hover over `ESLintOptions`!